Modules/_pickle.c

```diff
@@ -360,6 +360,81 @@ _Pickle_FastCall(PyObject *func, PyObject *obj)
 
 /*************************************************************************/
 
+static int
```

Comment:
I wonder if the role of those three functions would be clearer if they took a `{PyObject *func, int unbound}` structure.